docs/fluvio/concepts/batching.mdx

```diff
@@ -6,10 +6,87 @@ title: "Batching"
 - `compression`: Compression algorithm used by the producer to compress each batch before sending it to the SPU. Supported compression algorithms are none, gzip, snappy and lz4.
+- `linger`:  Time to wait before sending batches to the server that have not reached maximum batch size. Defaults to 100 ms.
 
-In general, each one of these configurations has a benefit and a potential drawback. For instance, with the compression algorithm, it is a trade-off between disk usage in the server and CPU usage in the producer and the consumer for compression and decompression. Typically, the compression ratio is improved when the payload is large, therefore a larger `batch_size` could be used to improve the compression ratio. A `linger` equals `0` means that each record is sent as soon as possible. A `linger` time larger than zero introduces latency but improves throughput.
 
-The ideal parameters for the `batch_size`, `linger` and `compression` depend on your application needs.
+# Trade-offs and Considerations
+
+Each configuration offers a balance of benefits and trade-offs:
+
+- `max_request_size`: Allows the producer to send larger records, but may increase the risk of network congestion and latency. 
+- `batch_size`: Larger value can reduce the number of requests sent to the server, but may increase latency. Smaller value can reduce latency, but may increase the number of requests.
+- `compression`: Helps reduce storage space at the cost of increased CPU usage for compressing and decompressing data. Larger batch_size values generally improve compression ratios, especially with larger payloads.
+- `linger`: A value of 0 sends records immediately, minimizing latency but potentially reducing throughput. Higher values introduce some delay but can enhance throughput by enabling more efficient batching.
+
+The ideal parameters for the `max_request_size`, `batch_size`, `linger` and `compression` depend on your application needs.
+
+# Example Scenarios
+
+Before testing, create a topic and generate a large data file:
+
+```bash
+fluvio topic create example-topic
+printf 'This is a sample line. ' | awk -v b=500000 '{while(length($0) < b) $0 = $0 $0}1' | cut -c1-500000 > large-data-file.txt
+```
+
+### Max Request Size
+
+`max_request_size` will define the maximum size of a message that can be sent by the producer. If a message exceeds this size, Fluvio will throw an error. Even if it's a message with only one record or a batch of them.
+
+```bash
+fluvio produce example-topic --max-request-size 16384 --file large-data-file.txt --raw
+```
+
+Will be displayed the following error:
+
+```bash
+the given record is larger than the max_request_size (16384 bytes).
+```
+
+### Batch Size
+
+`batch_size` will define the maximum size of a batch of records that can be sent by the producer. If a record exceeds this size, Fluvio will process the record in a new batch without the `batch_size` as limit.
```

Comment:
will define the maximum size of a batch of records that the producer can send.  => `defines the cumulative size of all records sent in the same batch`

Reply:
Fixed!

Reply:
Fixed!